crates/subspace-farmer/src/dsn.rs

```diff
@@ -51,87 +55,43 @@ where
     DSN: DSNSync + Send + Sized,
     DSN::Stream: Unpin + Send,
     // On pieces might `break` with some result from the sync or just continue
-    OP: FnMut(FlatPieces, Vec<PieceIndex>) -> ControlFlow<anyhow::Result<()>> + Send + 'static,
+    OP: FnMut(FlatPieces, Vec<PieceIndex>) -> anyhow::Result<()> + Send + 'static,
```

Comment:
While looking at this, I'm always under the delusion that `Send + 'static` is applied to the result of `OP` instead of `OP` itself, just wondering if we can make it clearer on the first glance, perhaps:

```rust
    OP: FnMut(FlatPieces, Vec<PieceIndex>) -> anyhow::Result<()>,
    OP: Send + 'static,
```
or

```rust
    OP: Send + 'static + FnMut(FlatPieces, Vec<PieceIndex>) -> anyhow::Result<()>,
```

Reply:
Yeah, makes sense. Generally it is hard to confuse those things as closure return type is always specific, so no additional requirements can be specified to it